network/crl_test.go

```diff
@@ -0,0 +1,468 @@
+/*
+Copyright 2020, Cossack Labs Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"bytes"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"path"
+	"testing"
+	"time"
+)
+
+const (
+	// Default group, without intermediate certificates
+	TestCertPrefix          = "../tests/ssl"
+	TestCACertFilename      = "ca/ca.crt"
+	TestCertFilename        = "acra-writer/acra-writer.crt"
+	TestRevokedCertFilename = "acra-writer-revoked/acra-writer-revoked.crt"
+	TestCRLFilename         = "crl.pem"
+	TestOCSPCertFilename    = "ocsp-responder/ocsp-responder.crt"
+	TestOCSPKeyFilename     = "ocsp-responder/ocsp-responder.key"
+
+	// TODO add certificates for this group, fix paths, uncomment related code
+	// A different group, uses intermediate CA
+	// TestCertPrefix3          = "../tests/ssl/group-with-intermediate"
+	// TestCACertFilename3      = "ca/ca.crt"
+	// TestICACertFilename3     = "intermediate-ca/intermediate-ca.crt"
+	// TestCertFilename3        = "acra-writer/acra-writer.crt"
+	// TestRevokedCertFilename3 = "acra-writer-revoked/acra-writer-revoked.crt"
+	// TestCRLFilename3         = "intermediate-ca/crl.pem"
+	// TestOCSPCertFilename3    = "intermediate-ocsp-responder/intermediate-ocsp-responder.crt"
+	// TestOCSPKeyFilename3     = "intermediate-ocsp-responder/intermediate-ocsp-responder.key"
+
+	// Make sure you use either of these groups and don't mix their components
+)
+
+type TestCertGroup struct {
+	prefix string
+
+	ca  string
+	ica string // may be empty (missing)
```

Comment:
IMO, `intermediateCA` will be more obvious when someone reads this code half a year from now.